### PR TITLE
31335-docs: Create cis-compliance.md

### PR DESCRIPTION
**Pull request by jonquil2002**
SC: https://app.shortcut.com/replicated/story/31335/cis-benchmark-compliance-remediation

Preview: https://deploy-preview-745--kurlsh-staging.netlify.app/docs/install-with-kurl/cis-compliance

### CODE REVIEW

**Comment by jonquil2002** [comment]
@JustinN-Replicated @jala-dx @emosbaugh - Please do not merge this until I make any necessary changes from you and get a peer review from Paige. Thanks!

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:11]: Should this be moved under known limitations?

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:17]:
```suggestion
This YAML file example shows a valid specification for CIS compliance:
```

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:42]: How about using the latest spec here?

```suggestion
  kubernetes: 
    version: "1.23.x"
  weave: 
    version: "2.6.x"
  contour: 
    version: "1.20.x"
  prometheus: 
    version: "0.53.x"
  registry: 
    version: "2.7.x"
  containerd: 
    version: "1.4.x"
  ekco: 
    version: "latest"
  minio: 
    version: "2020-01-25T02-50-51Z"
  longhorn: 
    version: "1.2.x"
```

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:47]:
```suggestion
* The [EKCO add-on](/docs/add-ons/ekco) is required to use this feature.
```

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:48]:
```suggestion
* This feature currently only works with the [Kubernetes (kubeadm) add-on](https://kurl.sh/docs/add-ons/kubernetes).
```

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:50]:
```suggestion
* Kubelet will no longer attempt to modify non-standard kernel flags which may block the Kubelet from initializing causing the installation to fail.
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (pavansokkenagaraj) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:42]: latest is deprecated and not used anymore.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (jala-dx) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:47]: Ecko (0.19.0) and above is required to use this flag.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:42]: @pavansokkenagaraj can you suggest a different spec?

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (jonquil2002) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:11]: OK.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (jonquil2002) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:50]: Added, slightly modified.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:40]: This spec still needs to change. I don't think we should encourage people to use antrea and rook.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (JustinN-Replicated) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:13]: Is there a reason why we don't open the link in a new tab since it is external to our docs? so something like this

<a href="https://www.cisecurity.org/benchmark/kubernetes" target="_blank">CIS benchmark information</a>.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (JustinN-Replicated) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:9]: @jala-dx @emosbaugh can we put more details here? I had thought we would outline all of the settings that were behind the flag.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (JustinN-Replicated) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:43]: @jala-dx We likely need something under known limitations about upgrading existing kURL to using this flag is unsupported. Below is first draft:

* Upgrading an existing kURL installation to including `kurl.cisCompliance` field to `true` is unsupported as the settings that are changed may introduce adverse effects.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (jonquil2002) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:13]: Apparently opening links in new tabs isn't supported in classic Markdown. You have to Kramdown to do it, and I'm not sure if that would break our Netlify previews. I checked some other external links in this repository and they do not open in new tabs either.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (JustinN-Replicated) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:40]: Could we just take the current one from kurl.sh and add kots to it? 

apiVersion: "cluster.kurl.sh/v1beta1"
kind: "Installer"
metadata: 
  name: "latest"
spec: 
  kurl:
    cisCompliance: true
  kotsadm:
    version: "latest"
  kubernetes: 
    version: "1.23.x"
  weave: 
    version: "2.6.x"
  contour: 
    version: "1.20.x"
  prometheus: 
    version: "0.53.x"
  registry: 
    version: "2.7.x"
  containerd: 
    version: "1.4.x"
  ekco: 
    version: "latest"
  minio: 
    version: "2020-01-25T02-50-51Z"
  longhorn: 
    version: "1.2.x"

File: src/markdown-pages/install-with-kurl/ipv6.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/ipv6.md:29]: In limitations it says antrea is the only supported ipv6 cni

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (jonquil2002) [reply src/markdown-pages/install-with-kurl/cis-compliance.md]: @JustinN-Replicated Just committed these changes, though I'm not sure why the preview isn't rebuilding.

**Comment by JustinN-Replicated** [comment]
@jonquil2002 are you removing the ipv6 portion or are we OK with that staying in this PR? I made the minor adjustment needed that Ethan had pointed out.

File: src/markdown-pages/install-with-kurl/ipv6.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/ipv6.md:21]: @JustinN-Replicated do you want to change the name back to ipv6 here?

Comment (jonquil2002) [reply src/markdown-pages/install-with-kurl/ipv6.md:21]: @emosbaugh @JustinN-Replicated I'm ok with keeping Justin's changes to IPv6.

**Review by jonquil2002 — COMMENTED** [review]
Justin made these changes.

File: src/markdown-pages/install-with-kurl/ipv6.md
Comment (JustinN-Replicated) [reply src/markdown-pages/install-with-kurl/ipv6.md:21]: Yep, sorry about that. Will adjust now

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:25]:
```suggestion
`--protect-kernel-defaults` is set to `true`.
```

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:13]: I feel like these configuration sections are missing a whole lot eg audit log and profiling settings. what are we looking to capture here?

Comment (JustinN-Replicated) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:13]: I can't find the comment, but I had asked in an earlier comment that we should be noting what is changed or turned on when this flag is used. In particular, I would imagine all of these settings behind the flag should be mentioned/logged so people know what the flag is doing.

Comment (emosbaugh) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:13]: oh. i get it. these are only the ones behind the flag. lgtm

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:64]:
```suggestion
* The [EKCO add-on](/docs/add-ons/ekco) v0.19.0 and later is required to use this feature.
```

**Review by emosbaugh — APPROVED** [review]

**Comment by jonquil2002** [comment]
@JustinN-Replicated Lmk when/if this ready for your approval. After you do that, I'll add Paige to do the final editorial review. Then you can merge when the release is ready.

**Comment by JustinN-Replicated** [comment]
@jonquil2002 from the PR, it doesn't look like I am approved reviewer. Having said that, AFAIK this one is good to go; I think we hashed out all questions yesterday and all changes are incorporated.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (paigecalvert) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:9]: Active voice recommendation:
"The kURL installer can be configured to be Center for Internet Security (CIS) compliant. This is an opt-in feature that is configured in the kURL specification by setting the `kurl.cisCompliance` field to `true`." -->

"You can configure the kURL installer to be Center for Internet Security (CIS) compliant. You can opt-in to this feature by setting the `kurl.cisCompliance` field to `true` in the kURL installer specification."

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (paigecalvert) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:62]: Perhaps this is "Requirements and Limitations" rather than Known Limitations? The EKCO add-on bullet looks more like a requirement.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (paigecalvert) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:19]: Is there more detail we can add to "as appropriate", or is this assumed knowledge? I suppose it is set to the certificate authority that the vendor uses for their API server configuration? (or something like that?)

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (paigecalvert) [inline src/markdown-pages/install-with-kurl/cis-compliance.md:11]: Active voice recommendation:

When you set `cisCompliance` to `true`, kURL changes the following settings: 

Also, I'm lacking background so of course ignore if this isn't relevant, but I'm wondering if "changes" is the right verb here. Do these settings already have values that are being changed? Or, is the installer setting some of the values for the first time?

**Review by paigecalvert — APPROVED** [review]
Looks great, added a few suggestions and questions

**Review by jala-dx — APPROVED** [review]

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (jonquil2002) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:9]: Addressed in this PR: https://github.com/replicatedhq/kurl.sh/pull/763

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (jonquil2002) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:19]: I checked with Jalaja, and this is assumed knowledge.

File: src/markdown-pages/install-with-kurl/cis-compliance.md
Comment (jonquil2002) [reply src/markdown-pages/install-with-kurl/cis-compliance.md:11]: Resolved in https://github.com/replicatedhq/kurl.sh/pull/763

**Pull request merged**
